Automattic/jetpack
# [RNMobile] Tiled Gallery Block: Aspect Ratio for Square Layout

`Gutenberg Mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4017

This PR is one part of a wider effort to port the Jetpack Tiled Gallery block to Mobile Gutenberg. It will be merged into a feature branch, `rnmobile/add/tiled-gallery-block`.

For this specific PR, the purpose is to define the aspect ratio of images in the editor, so that they appear as squares.

#### Changes proposed in this Pull Request:

* With this PR, the images in the Tiled Gallery block will appear as squares in the editor by default.

#### Brief overview of code changes proposed in this Pull Request:

* [A new context, `imageCrop`, has been added to the index.js file](https://github.com/Automattic/jetpack/pull/21166/files#diff-025ac7803bd106aa93c59895bc274e4a297c5f27934dc461334ed2692cb045f9R299). This is an existing context in the image block, which [translates to `resizeMode=cover`](https://github.com/WordPress/gutenberg/pull/35089/files#diff-7126ae03000a022bcefa0d2a5fb03b6aca56ed4047e3a49d2131babf229cf6c1R633), which is necessary in order for us to be able to change the image's shape.
* [A `className` prop](https://github.com/Automattic/jetpack/pull/21166/files#diff-33587eede7ad93ef2a32666df26302b46dfd9d688d44ecf330efcb2e67606b27R109) is passed to the inner image blocks, which in turn will be passed to [the image block's `shapeStyle` prop](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/image/edit.native.js#L694) and used to pass styles. 
* The `is-style-squared` CSS selector is [defined directly in the Tiled Gallery's `styles.native.css` file](https://github.com/Automattic/jetpack/pull/21166/files#diff-df8a4982181830def37c42b47339a188260ac036e62ab4137834559771b8aee1R15) and its styles are the ones being passed to the inner image blocks, making use of the fact that [aspect ratio is supported by React Native](https://reactnative.dev/docs/layout-props#aspectratio).

#### Jetpack product discussion

Tiled Gallery project thread: p9ugOq-1Tb-p2 

#### Does this pull request change what data or activity we track or use?

No, it doesn't.

#### Testing instructions:






* With these branches checked out and the Gutenberg Mobile demo app running, add the Tiled Gallery block.
* Add a few images to the block.
* Observe the shape of the images and verify that they're squares.

A few ideas for testing:

* Add varying numbers of images to ensure the images are squares even when there are different numbers per row.
* Switch between portrait and landscape to check for any inconsistencies.
* Test the core image and gallery blocks for any regressions.

#### Screenshots

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/134989742-1b86fb15-1699-4b21-ae22-612527151a21.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/134991240-cfeac12b-4dd7-4466-b3ca-55948d09149a.png" width="100%"> |
